### PR TITLE
Removing Lagrange spatial discretizations and cell mappings.

### PR DESCRIPTION
**Pull request by wdhawkins**

### CODE REVIEW

**Review by ragusa — APPROVED** [review]

**Review by ragusa — COMMENTED** [review]
great. in deal.ii, the tradition is fe_data and fe_face_data (IIRC). Here, you have opted for fe_vol_data and fe_srf_data. it's fine.

I would approve but format tests are failing.

File: framework/field_functions/field_function_grid_based.cc [hunk 1]
```diff
@@ -155,8 +154,7 @@ FieldFunctionGridBased::MakeSpatialDiscretization(const InputParameters& params)
     if (cs == "spherical") q_order = QuadratureOrder::FOURTH;
   }
 
-  if (sdm_type == "PWLC")
-    return PWLC::New(*grid_ptr, q_order, cs_type);
+  if (sdm_type == "PWLC") return PWLC::New(*grid_ptr, q_order, cs_type);
   else if (sdm_type == "PWLD")
     return PWLD::New(*grid_ptr, q_order, cs_type);
```
Comment (ragusa) [inline framework/field_functions/field_function_grid_based.cc:159]: this clang format is ugly. it was better before! I think you can halt clang-format. You can just add:
// clang-format off
    unformatted_code 
// clang-format on
to keep as it was before. We have this in several places

File: test/framework/math/spatial_discretization/dg/sdm_test_01_discontinuous.cc [hunk 2]
```diff
@@ -65,8 +65,7 @@ math_SDM_Test02_DisContinuous(const InputParameters& input_parameters)
 
   {
     using namespace opensn;
-    if (sdm_type == "PWLD") 
-      sdm_ptr = PieceWiseLinearDiscontinuous::New(grid);
+    if (sdm_type == "PWLD") sdm_ptr = PieceWiseLinearDiscontinuous::New(grid);
     else
       ChiInvalidArgument("Unsupported sdm_type \"" + sdm_type + "\"");
```
Comment (ragusa) [inline test/framework/math/spatial_discretization/dg/sdm_test_01_discontinuous.cc:70]: same

File: test/framework/math/spatial_discretization/cg/sdm_test_01_continuous.cc [hunk 3]
```diff
@@ -50,8 +50,7 @@ math_SDM_Test01_Continuous(const InputParameters& input_parameters)
   bool is_DG = false;
   {
     using namespace opensn;
-    if (sdm_type == "PWLC")
-      sdm_ptr = PieceWiseLinearContinuous::New(grid);
+    if (sdm_type == "PWLC") sdm_ptr = PieceWiseLinearContinuous::New(grid);
     else
       ChiInvalidArgument("Unsupported sdm_type \"" + sdm_type + "\"");
```
Comment (ragusa) [inline test/framework/math/spatial_discretization/cg/sdm_test_01_continuous.cc:55]: same,

Comment (ragusa) [inline test/framework/math/spatial_discretization/cg/sdm_test_01_continuous.cc:55]: @andrsd : should there sitll be a chiXXXXX at this stage?

**Review by ragusa — COMMENTED** [review]
a few comments left. easy fix.

**Comment by wdhawkins** [comment]
> great. in deal.ii, the tradition is fe_data and fe_face_data (IIRC). Here, you have opted for fe_vol_data and fe_srf_data. it's fine.
> 
> I would approve but format tests are failing.

If we prefer to go with fe_data and fe_face_data, that's a trivial change at this point. I can easily do that.

File: framework/field_functions/field_function_grid_based.cc (hunk 1 above)
Comment (zhardy-lanl) [reply framework/field_functions/field_function_grid_based.cc]: I'm not totally sure of the clang-format option, but I've felt this way a few times before on this. I wouldn't be opposed to making all if statements go to the next line.

**Comment by zhardy-lanl** [comment]
> > great. in deal.ii, the tradition is fe_data and fe_face_data (IIRC). Here, you have opted for fe_vol_data and fe_srf_data. it's fine.
> > I would approve but format tests are failing.
> 
> If we prefer to go with fe_data and fe_face_data, that's a trivial change at this point. I can easily do that.

Since we have `Volumetric` and `Surface` FE data, I think the current version is okay. Using `face` introduces a small inconsistency. For some reason `srf` just looks a little weird to me, but I don't have a good alternative unless we rename `Volumetric` and `Surface` FE data to `Cell` and `Face`, respectively. I'm agnostic on that.

**Review by zhardy-lanl — APPROVED** [review]

File: test/framework/math/spatial_discretization/cg/sdm_test_01_continuous.cc (hunk 3 above)
Comment (andrsd) [reply test/framework/math/spatial_discretization/cg/sdm_test_01_continuous.cc]: Yes. Issue #65 will take care of the names and probably more...

File: framework/field_functions/field_function_grid_based.cc (hunk 1 above)
Comment (andrsd) [reply framework/field_functions/field_function_grid_based.cc]: There is one for that. I can send a PR with the change.

**Review by andrsd — APPROVED** [review]

**Pull request merged**